script/lib/createBuildDescriptor.js

```diff
@@ -0,0 +1,19 @@
+const fs = require("fs");
+
+function createBuildDescriptor(config) {
+    let yarnLockContent = fs.readFileSync("yarn.lock");
+
+    let dir = "dist/";
+
+    if (config["dirs"] && config["dirs"]["dist"]) {
+        dir = config["dirs"]["dist"];
+        if (!dir.endsWith("/")) dir += "/";
+    }
+
+    fs.writeFile(dir + "app_descriptor.lock", yarnLockContent, (err) => {
```

Comment:
Here I suggest to keep the name `yarn.lock` since that's the content.
Then you add the file with `app_version.xxx` with `git describe` output.